plugins/pulp_rpm/plugins/importers/yum/config_validate.py

```diff
@@ -25,3 +46,69 @@ def validate(config):
             msg += failure_message + '\n'
         msg = msg.rstrip()  # remove the trailing \n
         return False, msg
+
+
+def _validate_allow_keys(repo, config, error_messages):
```

Comment:
The use case you describe is valid, but I think it's a separate story on its own. As I imagined this story, removing a key when there are units_in_repo > 0 we would also need to then remove units signed by that key too. Since we're just getting started with this feature I propose that we don't try to handle that case. Writing that as a separate story would be great.

The user could workaround this limitation on their own by creating a second repo with the uncompromised key and doing a copy all operation.

Also regarding partial updates where some of the update data is accepted but other update data is not, generally I think we should not do that. When applying a change I would feel most comfortable if either the entire update applies or none of it applies.